kernel/msg_q.c

```diff
@@ -136,6 +136,31 @@ _SYSCALL_HANDLER(k_msgq_put, msgq_p, data, timeout)
 }
 #endif
 
+int _impl_k_msgq_get_attrs(struct k_msgq *q, struct k_msgq_attrs *attrs)
+{
+	s32_t ret = -EINVAL;
+
+	if (attrs != NULL) {
```

Comment:
Checking for NULL arguments isn't really a zephyr thing.  Why not put the checking in the POSIX layer (or userspace, where you get it for free) and not duplicate it here?

Seems like this would be a clearer API without the error return code and without the check.  There is no circumstance where a valid k_msgq will fail to have attribute values.

Reply:
There's no need to check this in the handler function either: _SYSCALL_MEMORY_WRITE() should be sufficient.

Reply:
Agree,  I will remove the check for attr from here.